src/phoenix/server/api/mutations/dataset_mutations.py

```diff
@@ -406,6 +517,33 @@ def _span_attribute(semconv: str) -> Any:
     return attribute_value.label(semconv.replace(".", "_"))
 
 
+def _to_orm_revision(
+    *,
+    previous_revision: models.DatasetExampleRevision,
+    patch: DatasetExamplePatch,
+    example_id: int,
+    version_id: int,
+) -> Dict[str, Any]:
+    """
+    Creates a new revision from a previous revision and a patch. The output is a
+    dictionary suitable for insertion into the database using the sqlalchemy
+    bulk insertion API.
+    """
+
+    db_rev = models.DatasetExampleRevision
+    return {
+        str(db_column.key): patch_value
+        for db_column, patch_value in (
+            (db_rev.dataset_example_id, example_id),
+            (db_rev.dataset_version_id, version_id),
+            (db_rev.input, patch.input or previous_revision.input),
+            (db_rev.output, patch.output or previous_revision.output),
+            (db_rev.metadata_, patch.metadata or previous_revision.metadata_),
```

Comment:
How would this work if I want to delete the metadata?

Reply:
`input`, `output`, and `metadata` are non-optional JSON types, so you would need to pass in an empty dictionary for the metadata.

Reply:
i was referring to `patch.metadata or existing_revision.metadata_`, which would not allow deletion of metadata because of the `or`

Reply:
Good catch. Addressed in 07f6430ace5273f2d728d839c4e2f97834126640.